### PR TITLE
Adding support to nuke selected region

### PR DESCRIPTION
**Pull request by saurabh-hirani**
- Adds --region flag to select specific regions.
- Updated README.md

### CODE REVIEW

**Comment by saurabh-hirani** [comment]
This also improves on the UX by:

1. Calling out the regions in the first log message. 
2. Not mixing ``` Skipping region ``` log message with ```Checking region``` as the user is interested only in the target region.
3. Adding current_n / total_n so that the user is aware of the retrieval progress.


Instead of 

```
go run main.go aws  --exclude-region eu-north-1 --exclude-region ap-south-1 --exclude-region eu-west-1 --exclude-region ap-northeast-2                            

INFO[2019-09-21T17:10:38+00:00] Retrieving all active AWS resources
INFO[2019-09-21T17:10:38+00:00] Skipping region: eu-north-1
INFO[2019-09-21T17:10:38+00:00] Skipping region: ap-south-1
INFO[2019-09-21T17:10:38+00:00] Checking region: eu-west-3
INFO[2019-09-21T17:10:47+00:00] Checking region: eu-west-2                   
INFO[2019-09-21T17:11:00+00:00] Skipping region: eu-west-1                   
INFO[2019-09-21T17:11:00+00:00] Skipping region: ap-northeast-2              
INFO[2019-09-21T17:11:00+00:00] Checking region: ap-northeast-1       
.
.
```

We now have 

```
go run main.go aws  --exclude-region eu-north-1 --exclude-region ap-south-1 --exclude-region eu-west-1 --exclude-region ap-northeast-2                            

INFO[2019-09-21T17:11:23+08:00] Retrieving active AWS resources in [eu-west-3, eu-west-2, ap-northeast-1, sa-east-1, ca-central-1, ap-southeast-1, ap-southeast-2, eu-central-1, us-east-1, us-east-2, us-west-1, us-west-2]
INFO[2019-09-21T17:11:23+00:00] Checking region [1/12]: eu-west-3
INFO[2019-09-21T17:11:32+00:00] Checking region [2/12]: eu-west-2
INFO[2019-09-21T17:11:42+00:00] Checking region [3/12]: ap-northeast-1       
INFO[2019-09-21T17:11:46+00:00] Checking region [4/12]: sa-east-1            
INFO[2019-09-21T17:11:58+00:00] Checking region [5/12]: ca-central-1         
.
.
.
```

4. Makes ```--region``` and ```--exclude-region``` mutually exclusive.

```
$ go run main.go aws --region us-east-1 --exclude-region us-west-1      
                     
ERRO[2019-09-21T17:59:25+00:00] --region and --exclude-region are mutually exclusive  error="--region and --exclude-region are mutually exclusive"
exit status 1
```
5. Improves function signature of  ```GetAllResources```  from

```
func GetAllResources(regions []string, excludedRegions []string, excludeAfter time.Time, resourceTypes []string) (*AwsAccountResources, error)                      
```

to

```
func GetAllResources(targetRegions []string, excludeAfter time.Time, resourceTypes []string) (*AwsAccountResources, error)                            
```

**Comment by brikis98** [comment]
@Etiene Could you review?

File: commands/cli.go [hunk 1]
```diff
@@ -111,24 +115,55 @@ func awsNuke(c *cli.Context) error {
 	if err != nil {
 		return errors.WithStackTrace(err)
 	}
+
+	selectedRegions := c.StringSlice("region")
 	excludedRegions := c.StringSlice("exclude-region")
 
+	if len(selectedRegions) > 0 && len(excludedRegions) > 0 {
```
Comment (Etiene) [inline commands/cli.go]: Logic seems good to me, but I wonder, for code cleanliness purposes, if this region filtering process shouldn't happen in an auxiliary function?

Comment (saurabh-hirani) [reply commands/cli.go]: Thanks for the feedback @Etiene  - makes sense. I moved the region filtering process to a separate function ```aws.GetTargetRegions``` which in turn also helped to create tests for the same in ```aws_test.go``` as ```TestGetTargetRegions```

**Comment by saurabh-hirani** [comment]
@Etiene  with the updated code as per the feedback, when I tested ```aws_test.go```'s test function ```TestGetTargetRegions``` I got the error:

```
~/go/packages/src/github.com/gruntwork-io/cloud-nuke/aws/mocks/EC2API.go:389:72: undefined: ec2.ApplySecurityGroupsToClientVpnTargetNetworkInput
~go/packages/src/github.com/gruntwork-io/cloud-nuke/aws/mocks/EC2API.go:389:128: undefined: ec2.ApplySecurityGroupsToClientVpnTargetNetworkOutput
.
.
```

which points to an older version of https://github.com/gruntwork-io/cloud-nuke/blob/master/aws/mocks/EC2API.go 
I regenerated the same file by following - https://github.com/gruntwork-io/cloud-nuke/blob/master/aws/ec2_unit_test.go#L7 as 

```
mockgen -source vendor/github.com/aws/aws-sdk-go/service/ec2/ec2iface/interface.go -destination aws/mocks/EC2API.go
```

and the tests **worked**. The new ```EC2API.go``` has significant changes from the older ones. I have pasted a diff here - https://gist.github.com/saurabh-hirani/fe8419c08d60fe734952b5a36972194f

Should this be tracked in a separate issue? I didn't want to pollute the existing PR with the same and wanted your advice on it.

File: commands/cli.go (hunk 1 above)
Comment (saurabh-hirani) [reply commands/cli.go]: Hi @Etiene - just wanted to check if this is fine and can be merged? Thanks.

**Comment by saurabh-hirani** [comment]
@brikis98 @Etiene - can you please check and let me know if this is good to merge?

**Comment by autero1** [comment]
Sorry for the delay, @saurabh-hirani . I can take a look at this.

**Comment by saurabh-hirani** [comment]
Thanks @autero1

**Comment by autero1** [comment]
I re-ran the tests with your latest commits and the tests passed. Unfortunately I didn't get to make a final pass on this PR, but will try and finalize this first thing Monday morning.

**Comment by saurabh-hirani** [comment]
Thanks for the review and merge @autero1!

**Comment by autero1** [comment]
Still waiting for the tests to pass before I release.

**Comment by autero1** [comment]
https://github.com/gruntwork-io/cloud-nuke/releases/tag/v0.1.11

**Pull request merged**